src/main/java/com/uwaterloo/connect/controller/ActivityController.java

```diff
@@ -0,0 +1,92 @@
+package com.uwaterloo.connect.controller;
+
+import com.uwaterloo.connect.model.ActivityRequest;
+import com.uwaterloo.connect.repository.ActivityRepository;
+import com.uwaterloo.connect.service.ActivityService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import static com.uwaterloo.connect.Constants.ActivityEndpointURLs.*;
+import static com.uwaterloo.connect.Constants.Constants.ERROR;
+import static com.uwaterloo.connect.Constants.Constants.SUCCESS;
+
+@RestController
+@RequestMapping("/activity")
+public class ActivityController {
+    @Autowired
+    ActivityRepository activityRepository;
+    @Autowired
+    ActivityService activityService;
+
+    @GetMapping(ADD_ACTIVITY)
+    public ResponseEntity<String> addActivity(@RequestBody ActivityRequest activityRequest){
+        String result = activityService.createActivity(activityRequest);
+        if(SUCCESS.equals(result)){
```

Comment:
Not a blocker, but we can have a common class/method, since these couple of lines are common across the class

Reply:
Fixed in local